ugvc/mrd/balanced_strand_utils.py

```diff
@@ -1188,6 +1189,73 @@ def plot_trimmer_histogram(
     return axs
 
 
+def flatten_strand_ratio_category(h5_file: str, key: str) -> pd.DataFrame:
+    df = pd.read_hdf(h5_file, key)
+    df["index_col"] = df.index
+    df.reset_index(drop=True, inplace=True)
+    df1 = df.melt(id_vars=["index_col"])
+    df1[""] = df1["index_col"] + "_" + df1["variable"]
+    df2 = df1[["", "value"]]
+    df2.set_index("", inplace=True)
+    df2 = df2.T
+    df2.reset_index(drop=True, inplace=True)
+    return df2
+
+
+def flatten_metrics(h5_file: str, key: str) -> pd.DataFrame:
+    df = pd.read_hdf(h5_file, key)
+    df.index.names = [""]
+    df = df.T
+    df.reset_index(drop=True, inplace=True)
+    return df
+
+
+def convert_h5_to_papyrus_json(h5_file: str, output_json: str) -> str:
+    """
+    Convert a statistics HDF5 file to a Papyrus JSON file
+    """
+    with pd.HDFStore(h5_file, "r") as store:
+        h5_file_keys = store.keys()
+    flatten_dfs = {}
+
+    # strand_ratio_category_counts
+    flatten_dfs["strand_ratio_category_counts"] = flatten_strand_ratio_category(
+        h5_file, "/strand_ratio_category_counts"
+    )
+
+    # strand_ratio_category_norm
+    flatten_dfs["strand_ratio_category_norm"] = flatten_strand_ratio_category(h5_file, "/strand_ratio_category_norm")
+
+    # stats_shortlist
+    flatten_dfs["stats_shortlist"] = flatten_metrics(h5_file, "/stats_shortlist")
+
+    # sorter stats
+    flatten_dfs["sorter_stats"] = flatten_metrics(h5_file, "/sorter_stats")
+
+    # category consensus
+    if "/df_category_concordance" in h5_file_keys:
+        flatten_dfs["category_consensus"] = flatten_metrics(h5_file, "/df_category_consensus")
+
+    # category concordance
+    if "/df_category_concordance" in h5_file_keys:
+        df_category_concordance = pd.read_hdf(h5_file, "/df_category_concordance")
+        df_category_concordance.index = df_category_concordance.index.to_flat_index()
+        df_category_concordance.rename(columns={"count": 0}, inplace=True)
+        df_category_concordance = df_category_concordance.T
+        flatten_dfs["category_concordance"] = df_category_concordance
+
+    # wrap all in one json
+    root_element = "metrics"
+    new_json_dict = {root_element: {}}
+    for key, df in flatten_dfs.items():
+        df_to_json = df.to_json(orient="table")
+        json_dict = json.loads(df_to_json)
+        new_json_dict[root_element][key] = json_dict
```

Comment:
I think it can be done in a one liner if you want
`new_json_dict[root_element] = {key: json.loads(df.to_json(orient="table")) for key, df in flatten_dfs.items()}`